compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -1242,14 +1242,15 @@ impl<'hir> LoweringContext<'_, 'hir> {
     }
 
     fn lower_expr_asm(&mut self, sp: Span, asm: &InlineAsm) -> hir::ExprKind<'hir> {
-        if self.sess.asm_arch.is_none() {
+        if self.sess.asm_arch.is_none() && !self.sess.opts.actually_rustdoc {
             struct_span_err!(self.sess, sp, E0472, "asm! is unsupported on this target").emit();
         }
         if asm.options.contains(InlineAsmOptions::ATT_SYNTAX)
             && !matches!(
                 self.sess.asm_arch,
                 Some(asm::InlineAsmArch::X86 | asm::InlineAsmArch::X86_64)
             )
+            && !self.sess.opts.actually_rustdoc
         {
             self.sess
                 .struct_span_err(sp, "the `att_syntax` option is only supported on x86")
```

Comment:
Why is this check done so early? I would expect it to be in codegen somewhere, or at least in MIR lowering.